core/src/avm1/value.rs

```diff
@@ -206,7 +163,7 @@ impl<'gc> Value<'gc> {
         activation: &mut Activation<'_, 'gc, '_>,
     ) -> Result<Value<'gc>, Error<'gc>> {
         let result = match self {
-            Value::Object(object) => {
+            Value::Object(object) if object.as_display_object().is_none() => {
```

Comment:
In general, what are the semantics of the `is_display_object` check around these functions (before and after the PR)? I have a hunch this feels more like a temporary hack than a general rule.

Reply:
Currently `Value::Object` contains both `ScriptObject`s and `StageObject`s, but `StageObject`s behave more like primitive values than `ScriptObject`s. This check prevents calling `valueOf` on a MovieClip.

In the future we'll want a separate `Value::Object` and `Value::MovieClip` variants, to avoid confusion.